### PR TITLE
FIX-#5203: don't raise `AttributeError: 'list' object has no attribute '_query_compiler'` in `join` op

### LINKED ISSUE #5203: BUG: join raises `AttributeError: 'list' object has no attribute '_query_compiler'`
### Modin version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest released version of Modin.

- [X] I have confirmed this bug exists on the main branch of Modin. (In order to do this you can follow [this guide](https://modin.readthedocs.io/en/stable/getting_started/installation.html#installing-from-the-github-master-branch).)

### Reproducible Example

```python
import modin.pandas as pd
import numpy as np

df1 = pd.DataFrame(np.ones([2,4]), columns = ['a','b','c','d'])
df2 = pd.DataFrame(np.ones([2,4]), columns = ['a','b','c','d'])
df3 = pd.DataFrame(np.ones([2,4]), columns = ['a','b','c','d'])

df1.join([df2, df3], how="inner", on="a")
```

### Issue Description

Non-obvious error message. 

### Expected Behavior

In that case we should raise `ValueError: Joining multiple DataFrames only supported for joining on index` exception as pandas does.

### Error Logs

<details>

```python-traceback

File "test_join.py", line 8, in <module>
    df1.join([df2, df3], how="inner", on="a")
  File "C:\projects\modin\modin\logging\logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "C:\projects\modin\modin\_compat\pandas_api\latest\dataframe.py", line 172, in join
    return self._join(
  File "C:\projects\modin\modin\logging\logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "C:\projects\modin\modin\pandas\dataframe.py", line 1321, in _join
    other._query_compiler,
AttributeError: 'list' object has no attribute '_query_compiler'

```

</details>

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5203 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/test/dataframe/test_join_sort.py [hunk 1]
```diff
@@ -166,6 +166,17 @@ def test_join(test_data, test_data2):
...
 
+def test_join_5203():
+    df1 = pd.DataFrame(np.ones([2, 4]), columns=["a", "b", "c", "d"])
+    df2 = pd.DataFrame(np.ones([2, 4]), columns=["a", "b", "c", "d"])
+    df3 = pd.DataFrame(np.ones([2, 4]), columns=["a", "b", "c", "d"])
+    with pytest.raises(
+        ValueError,
+        match="Joining multiple DataFrames only supported for joining on index",
+    ):
+        df1.join([df2, df3], how="inner", on="a")
```
[the remark this answers is not in the document]
Comment (anmyachev) [reply modin/pandas/test/dataframe/test_join_sort.py:177]: we can, code is updated

**Review by dchigarev — DISMISSED** [review]

File: modin/pandas/dataframe.py [hunk 2]
```diff
@@ -1335,6 +1335,10 @@ def join(
         """
         Join columns of another ``DataFrame``.
         """
+        if on is not None and not isinstance(other, (Series, DataFrame)):
+            raise ValueError(
+                "Joining multiple DataFrames only supported for joining on index"
```
Comment (vnlitvinov) [inline modin/pandas/dataframe.py:1340]: is this error message consistent with what pandas reports?

File: modin/pandas/test/dataframe/test_join_sort.py [hunk 3]
```diff
@@ -166,6 +166,17 @@ def test_join(test_data, test_data2):
         df_equals(modin_join, pandas_join)
 
 
+def test_join_5203():
+    df1 = pd.DataFrame(np.ones([2, 4]), columns=["a", "b", "c", "d"])
+    df2 = pd.DataFrame(np.ones([2, 4]), columns=["a", "b", "c", "d"])
+    df3 = pd.DataFrame(np.ones([2, 4]), columns=["a", "b", "c", "d"])
+    with pytest.raises(
```
Comment (vnlitvinov) [inline modin/pandas/test/dataframe/test_join_sort.py:173]: I would suggest to verify that pandas produces the same error as well

File: modin/pandas/dataframe.py (hunk 2 above)
Comment (anmyachev) [reply modin/pandas/dataframe.py:1340]: yes

File: modin/pandas/test/dataframe/test_join_sort.py (hunk 3 above)
Comment (anmyachev) [reply modin/pandas/test/dataframe/test_join_sort.py]: done

**Review by dchigarev — APPROVED** [review]

**Pull request merged**